packages/status-js/src/client/chat.ts

```diff
@@ -202,32 +207,56 @@ export class Chat {
 
         // most recent message first
         while (--index >= 0) {
-          this.client.handleWakuMessage(wakuMessages[index])
+          const wakuMessage = wakuMessages[index]
+          // fixme!: type
+          wakuMessage._source = 'store'
+          // wakuMessage.proto._source = 'store'
+
+          this.client.handleWakuMessage(wakuMessage)
         }
       },
     })
 
-    // callback
+    // nothing found
+    if (!this.#messages.size) {
+      return
+    }
+
     // more not found
+    // todo?: what if on quering recieve update events which we'd like to callback on
```

Comment:
Could be another reason to `callback()` regardless if "more chat messages found", but also return `next` timestamp for the next fetch attempt.

Reply:
Those update events would be for messages we didn't receive yet – not need for callback.